timemachine/cpp/src/neighborlist.cu

```diff
@@ -356,8 +364,19 @@ template <typename RealType> int Neighborlist<RealType>::Y() const {
 
 template <typename RealType> int Neighborlist<RealType>::num_row_blocks() const { return ceil_divide(NR_, TILE_SIZE); }
 
+// max_ixn_count determines the number of tile to atom interaction counts. For each tile that interacts with another
+// it can have TILE_SIZE interactions. Note that d_ixn_count_ is only the number of tile interactions, and differs by a factor of TILE_SIZE
 template <typename RealType> int Neighborlist<RealType>::max_ixn_count() const {
-    return num_column_blocks() * num_row_blocks() * WARP_SIZE;
+    // The maximum number of interactions is that of all tiles interacting with all other tiles. Which
+    // is the upper triangular matrix.
+    // Use the maximum NxN matrix to allocate to allow supporting any combination of atom idxs
+    const int n_blocks = ceil_divide(max_size_, TILE_SIZE);
+    // Compute the size of the matrix without the diagonal
+    int max_tile_interactions = (n_blocks * n_blocks) - n_blocks;
+    // Divide by two to get the upper, then add back in the size of the diagonal
+    max_tile_interactions = ceil_divide(max_tile_interactions, 2) + n_blocks;
```

Comment:
nit: is `ceil_divide` needed here? I think $n^2 - n = n(n-1)$ should always be even?

Reply:
(then I think this could simplify to `max_tile_interactions = n_blocks * (n_blocks + 1) / 2`, but the way you have it commented is also nice)

Reply:
Simplified the logic and removed the unnecessary `ceil_divide` in https://github.com/proteneer/timemachine/pull/1163/commits/f7634b031660b8376ec7e104aed3097c3328793a